spec/models/manageiq/providers/nuage/network_manager/vcr_specs/targeted_refresh_spec.rb

```diff
@@ -0,0 +1,329 @@
+      :inventory_collections    => {
+        :saver_strategy => :default,
+      },
+    },
+    {
+      :inventory_object_refresh => true,
+      :allow_targeted_refresh   => true,
+      :inventory_collections    => {
+        :saver_strategy => :batch,
+        :use_ar_object  => true,
+      },
+    },
+    {
+      :inventory_object_refresh => true,
+      :allow_targeted_refresh   => true,
+      :inventory_collections    => {
+        :saver_strategy => :batch,
+        :use_ar_object  => false,
+      },
+    },
+    {
+      :inventory_object_saving_strategy => :recursive,
+      :inventory_object_refresh         => true,
+      :allow_targeted_refresh           => true,
+    }
+  ].freeze
+
+  before(:each) do
+    @ems = FactoryGirl.create(:ems_nuage_with_vcr_authentication, :port => 8443, :api_version => "v5_0", :security_protocol => "ssl-with-validation")
+  end
+
+  before(:each) do
+    userid   = Rails.application.secrets.nuage_network.try(:[], 'userid') || 'NUAGE_USER_ID'
+    password = Rails.application.secrets.nuage_network.try(:[], 'password') || 'NUAGE_PASSWORD'
+    hostname = @ems.hostname
+
+    # Ensure that VCR will obfuscate the basic auth
+    VCR.configure do |c|
+      # workaround for escaping host
+      c.before_playback do |interaction|
+        interaction.filter!(CGI.escape(hostname), hostname)
+        interaction.filter!(CGI.escape('NUAGE_NETWORK_HOST'), 'nuagenetworkhost')
+      end
+      c.filter_sensitive_data('NUAGE_NETWORK_AUTHORIZATION') { Base64.encode64("#{userid}:#{password}").chomp }
+    end
+  end
+
+  describe "targeted refresh" do
+    let(:network_group_ref)  { "e0819464-e7fc-4a37-b29a-e72da7b5956c" }
+    let(:security_group_ref) { "02e072ef-ca95-4164-856d-3ff177b9c13c" }
+    let(:cloud_subnet_ref1)  { "d60d316a-c1ac-4412-813c-9652bdbc4e41" }
+    let(:cloud_subnet_ref2)  { "debb9f88-f252-4c30-9a17-d6ae3865e365" }
+    let(:unexisting_ref)     { "unexisting-ems-ref" }
+
+    TARGET_REFRESH_SETTINGS.each do |settings|
+      context "with settings #{settings}" do
+        before(:each) do
+          stub_settings_merge(
+            :ems_refresh => {
+              :nuage_network => settings
+            }
+          )
+        end
+
+        describe "on empty database" do
+          it "will refresh cloud_subnet" do
+            cloud_subnet_target = ManagerRefresh::Target.new(
+              :manager     => @ems,
+              :association => :cloud_subnets,
+              :manager_ref => {:ems_ref => cloud_subnet_ref1}
+            )
+            test_targeted_refresh([cloud_subnet_target], 'cloud_subnet') do
+              assert_cloud_subnet_counts
+              assert_specific_cloud_subnet
+            end
+          end
+
+          it "will refresh cloud_subnet - object" do
+            cloud_subnet = FactoryGirl.create(:cloud_subnet_nuage, :ems_id => @ems.id, :ems_ref => cloud_subnet_ref1)
+            test_targeted_refresh([cloud_subnet], 'cloud_subnet') do
```

Comment:
right, so this one works, but MIQ will not allow this target through queue. The real integration test to prove it would be

example for containers, but it's all the same
```
target = ManagerRefresh::Target.new(
      :manager     => @ems,
      :association => :container_groups,
      :manager_ref => ems_ref,
      :options     => {
        :payload => notice['object'],
      }.to_json,
    )

    EmsRefresh.queue_refresh(target)
    expect(MiqQueue.where(:method_name => 'refresh').count).to eq 1
    refresh_job = MiqQueue.where(:method_name => 'refresh').first
    refresh_job.deliver
```

Reply:
But I suppose we can keep the current specs and add 1 more that will pass the target through queue

Reply:
I think the test in L71 above perfectly matches your description? 😉 
I mean, there is
```
it "will refresh cloud_subnet"
```
and
```
it "will refresh cloud_subnet - object"
```
already which covers what you describe here, I think.

Reply:
EDIT: ah, I thought you were referring to `ManagerRefresh::Target` vs providing the miq object directly. I see you actually meant to put to the queue, let me implement this as you suggest.

Reply:
yeah, so if you would put the ActiveRecord object into the Queue (which is the production workflow) you will see it fail, or ignore the target